### PR TITLE
fix: Same discount and access codes cannot be applied to two different events

### LINKED ISSUE #6027: Same Discount Code is not able to be applied for two different events
**Describe the bug**

Currently if you create two same discount codes for two events , You will see it is invalid for one of the event

**To Reproduce**

- Create Discount Code X for Event A
- Create Discount Code X for Event B
- Apply X to B 
- See Error

**Expected behavior**

Same Discount Should be able to apply

#### Screenshot

Screenshot 2019-06-07 at 5 49 25 PM

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #6027 

#### Short description of what this resolves:
Same discount and access code not able to apply to two different events.

#### Changes proposed in this pull request:
Checking discount and access code with event_id and access/discount code name, and changing the endpoint accordingly.

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/__init__.py [hunk 1]
```diff
@@ -613,7 +614,7 @@
 # Admin Statistics API
 api.route(AdminStatisticsSessionDetail, 'admin_statistics_session_detail', '/admin/statistics/sessions')
 api.route(AdminStatisticsEventDetail, 'admin_statistics_event_detail', '/admin/statistics/events')
-api.route(AdminStatisticsUserDetail, 'admin_statistics_user_detail', '/admin/statistics/users')
+api.route(AdminStatisticsUserDetail, 'admin_statistics_user_detail', '/admin/tistics/users')
```
Comment (iamareebjamal) [inline app/api/__init__.py]: What is tistics? Is it something millenials use instead of statistics?

File: tests/hook_main.py [hunk 2]
```diff
@@ -2804,15 +2804,11 @@ def discount_delete(transaction):
...
     """
-    GET /discount-codes/DC101
+    GET event_id/1/discount-code/DC101
     :param transaction:
     :return:
     """
     with stash['app'].app_context():
-        event = EventFactoryBasic()
-        db.session.add(event)
-        db.session.commit()
```
Comment (iamareebjamal) [inline tests/hook_main.py]: If you're going to remove this, how the discount code is going to be attached to the event?

File: tests/hook_main.py [hunk 3]
```diff
@@ -2804,15 +2804,11 @@ def discount_delete(transaction):
 @hooks.before("Discount Codes > Get Discount Code Detail using the code > Get Discount Code Detail")
 def discount_code_get_detail_using_code(transaction):
     """
-    GET /discount-codes/DC101
+    GET event_id/1/discount-code/DC101
```
Comment (iamareebjamal) [inline tests/hook_main.py]: And this is absolutely wrong. There is no route `event_id/` in the API. Not sure what you're trying to do

Comment (prateekj117) [reply tests/hook_main.py]: @iamareebjamal Had to write `event/`, made a mistake.

**Comment by prateekj117** [comment]
@iamareebjamal @uds5501 @mrsaicharan1 Please review.

**Comment by iamareebjamal** [comment]
See how simple it was. Had to make small changes. And how complicated you made it

**Comment by prateekj117** [comment]
@iamareebjamal Yaa !!!

**Pull request merged**